src/linker/implementation.py

```diff
@@ -47,10 +47,10 @@ def run(
             # (i.e. not 'local' computing environment) and so need to spin up a spark
             # cluster instead of relying on the implementation to do it in a container
             drmaa = get_slurm_drmaa()
-            spark_master_url, job_id = build_cluster(
+            spark_master_url, job_id = build_spark_cluster(
                 drmaa=drmaa,
                 session=session,
-                resources=self.resources,
+                resources=self.spark_resources,
```

Comment:
This may be out of scope for this PR (I'm not completely sure what the scope of this PR is), but I think you should send the config object here, not a dictionary. And then in `build_spark_cluster` call methods off the config object. Right now that method is depending on the very specific structure of the dictionary being passed, which leads to a tight coupling between these separate parts of the system.